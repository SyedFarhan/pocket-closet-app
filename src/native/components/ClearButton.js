import React from 'react';
import PropTypes from 'prop-types';
import { Text, Button } from 'native-base';


const ClearButton = ({ onClear }) => {
  return (
    <Button
      danger
      containerStyle={{ flex: 1 }}
      style={{ flex: 1, width: '100%' }}
      onPress={() => onClear()}
    >
      <Text>
        Clear
      </Text>
    </Button>
  );
};

ClearButton.propTypes = {
  onClear: PropTypes.func.isRequired,
};

export default ClearButton;
