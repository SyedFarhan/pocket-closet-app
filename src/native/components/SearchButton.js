import React from 'react';
import PropTypes from 'prop-types';
import { Text, Button } from 'native-base';


const SearchButton = ({ onSearch }) => {
  return (
    <Button
      containerStyle={{ flex: 1 }}
      style={{ flex: 1, width: '100%' }}
      onPress={() => onSearch()}
    >
      <Text>
        Search
      </Text>
    </Button>
  );
};

SearchButton.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default SearchButton;
