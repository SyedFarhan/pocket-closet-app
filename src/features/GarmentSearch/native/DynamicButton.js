import React from 'react';
import PropTypes from 'prop-types';

import SearchButton from './SearchButton';
import ClearButton from './ClearButton';

const DynamicButton = ({ isSearched, onSearch, onClear }) => {
  return (
    isSearched ? <ClearButton onClear={onClear} /> : <SearchButton onSearch={onSearch} />
  );
};

DynamicButton.propTypes = {
  isSearched: PropTypes.bool.isRequired,
  onSearch: PropTypes.func,
  onClear: PropTypes.func,
};

export default DynamicButton;
