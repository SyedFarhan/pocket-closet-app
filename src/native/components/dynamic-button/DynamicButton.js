import React from 'react';
import PropTypes from 'prop-types';

import SearchButton from '../search-button/SearchButton';
import ClearButton from '../clear-button/ClearButton';

const DynamicButton = ({ isSearched, onSearch, onClear }) => {
  return (
    isSearched ? <ClearButton onClear={onClear} /> : <SearchButton onSearch={onSearch} />
  );
};

DynamicButton.propTypes = {
  isSearched: PropTypes.bool.isRequired,
  onSearch: PropTypes.func,
  onClear: PropTypes.func,
};

export default DynamicButton;
