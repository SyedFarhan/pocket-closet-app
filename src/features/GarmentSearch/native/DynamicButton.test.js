/* global test */
import React from 'react';
import renderer from 'react-test-renderer';

import DynamicButton from './DynamicButton';
import ClearButton from './ClearButton';
import SearchButton from './SearchButton';

describe('DynamicButton', () => {
  let props;
  let mountedDynamicButton;
  const mountDynamicButtonWithProps = () => {
    mountedDynamicButton = mountedDynamicButton
      ? mountedDynamicButton
      : renderer.create(<DynamicButton {...props} />);
    return mountedDynamicButton;
  };

  const emptyFunction = () => {};

  const clearComponentAndProps = () => {
    props = {
      isSearched: undefined,
      onClear: undefined,
      onSearch: undefined,
    };
    mountedDynamicButton = undefined;
  };


  beforeEach(() => {
    clearComponentAndProps();
  });

  test('renders `SearchButton` when isSearched prop is false', () => {
    props.isSearched = false;
    props.onSearch = emptyFunction;
    mountDynamicButtonWithProps();

    const searchButton = mountedDynamicButton.root.findByType(SearchButton);
    expect(searchButton).toBeTruthy();
  });

  test('renders `ClearButton` when isSearched prop is true', () => {
    props.isSearched = true;
    props.onClear = emptyFunction;
    mountDynamicButtonWithProps();

    const clearButton = mountedDynamicButton.root.findByType(ClearButton);

    expect(clearButton).toBeTruthy();
  });

  // Snapshots
  test('Snapshot: renders SearchButton when isSearched prop is false', () => {
    const componentTree = renderer
      .create(<DynamicButton isSearched={false} onSearch={() => {}} />)
      .toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  test('Snapshot: renders ClearButton when isSearched prop is true', () => {
    const componentTree = renderer.create(<DynamicButton isSearched onClear={() => {}} />).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
