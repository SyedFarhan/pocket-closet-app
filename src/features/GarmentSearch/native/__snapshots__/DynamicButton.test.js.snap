// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicButton Snapshot: renders ClearButton when isSearched prop is true 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#d9534f",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 2,
      "flex": 1,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Clear
  </Text>
</View>
`;

exports[`DynamicButton Snapshot: renders SearchButton when isSearched prop is false 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#007aff",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 2,
      "flex": 1,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Search
  </Text>
</View>
`;
